### ISSUE TITLE
Translate to zh-tw

### ISSUE THREAD

**Issue by s8321414** [chubin/wttr.in#359]
I can translate wttr.in to zh-tw (Traditional Chinese for Taiwan), how can I do?

**Comment by chubin** [comment]
Yes, thank you very much! It would great. `zh-tw` is one of the most needed translations at the moment, and your help would be very appreciated.

To add the translation:


1. Fork the repo
2. Add the new translation here: https://github.com/chubin/wttr.in/tree/master/share/translations (you can use any of the existing translation files as the basis for your translation)
3. Add missing phrases to `translation.py`
4. Add the new `help.txt` (`zh-tw-help.txt`)
5. Do not forget to add your name to the list of the translators (`translation.txt`)

**Comment by s8321414** [comment]
I have opened a PR #361 :smiley:

**Comment by chubin** [comment]
Thank you for it! It is running live now. Please take a look at it:
```
$ curl zh-tw.wttr.in/Regensburg
$ curl zh-tw.wttr.in/:help
```
Do you like it? If everything is correct?

What about this?
```
│             早上             │             中午      └──────┬──────┘       晚間            │              晚              │
```
Is it correct? Or is not really `zh-tw`?

**Comment by s8321414** [comment]
Ah, evening should be 傍晚, should I make another PR for this?
All except this are fine :smiley:

**Comment by chubin** [comment]
You can't do a PR for it, because this part is not (yet) on GH. I will changed it.

Several other questions:
1. Is `zh-tw` = `simplified Chinese`?
2. If not, what code should be used for simpilfied Chinese?
3. What should be used for traditional Chinese? (see #207) `zh-hans`?
4. We would like to support zh-cn, zh-sg (or zh-hans?), zh-tw, zh-hk and zh-mo (zh-hant). See #83. What translation (from these ones) would you add first? May we use your translation as a base for the rest? Or it makes little sense?

**Comment by chubin** [comment]
I've fixed the 傍晚 thing; please check it: `curl zh-tw.wttr.in/Passau`

**Comment by s8321414** [comment]
> You can't do a PR for it, because this part is not (yet) on GH. I will changed it.
> 
> Several other questions:
> 
>     1. Is `zh-tw` = `simplified Chinese`?
> 
>     2. If not, what code should be used for simpilfied Chinese?
> 
>     3. What should be used for traditional Chinese? (see #207) `zh-hans`?
> 
>     4. We would like to support zh-cn, zh-sg (or zh-hans?), zh-tw, zh-hk and zh-mo (zh-hant). See #83. What translation (from these ones) would you add first? May we use your translation as a base for the rest? Or it makes little sense?

1. No, it is Traditional Chinese for Taiwan (or you can call it Taiwanese Mandarin)
2. zh-hans or zh-cn, should check what country you want to provide, but zh-hans is more general, and zh-cn is only for China (Mainland China you may call)
3. zh-hant, it is more general and different from zh-hans (for Simplified Chinese)
4. Well, you can use one of zh-hans locale for fallback to another zh-hans locales, it is same as zh-hant

**Comment by chubin** [comment]
That means that we need the following translations:

```
zh-hans
zh-cn
zh-sg

zh-hant
zh-tw
zh-hk
zh-mo
```

Right?
May I use zh-hant as an alias for `zh-tw`?
Or strictly speaking, `zh-hant` can differ from `zh-tw` in some cases?

What code should have Simiplified Chinese? `zh`?

**Comment by Vdragon** [comment]
As you may have noticed there are two distinguishing categories of language code, one by language(zh-hant, zh-hans) and one by region(the rest of them).

In general speaking region locale are more important as each region may have differences in many aspects of the language, as for the codes defined by language it should be an alias of the region code which has the most users of the language, and thus `zh-hant` should be an alias of `zh-tw`( which has most users using Traditional Chinese) and `zh-hans` should be an alias of `zh-cn` (which has most users using Simplified Chinese).

For `zh` locale(the Chinese language in general) it should be an alias of `zh-hans` since most users are using Simplified Chinese.

If the I18N framework supports language fallbacks the proper fallback priority should be region specific locale > language specific locale.

**Comment by s8321414** [comment]
Yes, just like @Vdragon says :smiley:

**Comment by s8321414** [comment]
By the way, how can we translate v2?

**Comment by chubin** [comment]
@s8321414 @Vdragon 

Hi guys! Could you please take a look at the v2 translation?
It is just a single line:

[lib/translations_v2.py](https://github.com/chubin/wttr.in/blob/master/lib/translations_v2.py)

**Comment by s8321414** [comment]
> @s8321414 @Vdragon
> 
> Hi guys! Could you please take a look at the v2 translation?
> It is just a single line:
> 
> [lib/translations_v2.py](https://github.com/chubin/wttr.in/blob/master/lib/translations_v2.py)

I created a [PR](https://github.com/chubin/wttr.in/pull/388) for this :smiley:

**Comment by chubin** [comment]
@s8321414 Thank you, Jeff! Merged
The `zh` trasnlation is still open. I hope @Vdragon (or someone else who is reading this) will help us with it

https://github.com/chubin/wttr.in/blob/master/lib/translations_v2.py#L67
